packages/react/src/components/Authenticator/Authenticator.tsx

```diff
@@ -22,7 +22,8 @@ export type AuthenticatorProps = AuthenticatorMachineOptions &
 
 // Helper component that sends init event to the parent provider
 function InitMachine({ children, ...data }) {
-  const { _send, route } = useAuthenticator();
+  // TODO: `INIT` event should be removed so that `_send` doesn't need to be extracted
+  const { _send, route } = useAuthenticator((context) => [context.route]);
```

Comment:
Just a minor callout that xstate docs recommends to define the selector functions outside out the component definition to optimize further. [Ref](https://xstate.js.org/docs/packages/xstate-react/#api) 

> // tip: optimize selectors by defining them externally when possible

Reply:
Thanks for the callout! I think it's referring to this `selector` we pass to `useSelector`:

https://github.com/aws-amplify/amplify-ui/blob/e3f1186df5bb936db14c0897b98e12cc3983aea1/packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx#L124

I had it inside the component because it needed a references to a `React.useRef`, but will keep this optimization in mind for our 3.0 changes.